src/ray/core_worker/core_worker.cc

```diff
@@ -281,13 +283,35 @@ void CoreWorker::ReportActiveObjectIDs() {
   heartbeat_timer_.async_wait(boost::bind(&CoreWorker::ReportActiveObjectIDs, this));
 }
 
-void CoreWorker::PromoteObjectToPlasma(const ObjectID &object_id) {
+void CoreWorker::PromoteToPlasmaAndGetOwnershipInfo(const ObjectID &object_id,
+                                                    TaskID *owner_id,
+                                                    rpc::Address *owner_address) {
   RAY_CHECK(object_id.IsDirectCallType());
   auto value = memory_store_->GetOrPromoteToPlasma(object_id);
   if (value != nullptr) {
     RAY_CHECK_OK(
         plasma_store_provider_->Put(*value, object_id.WithPlasmaTransportType()));
   }
+
+  auto has_owner = reference_counter_->GetOwner(object_id, owner_id, owner_address);
+  RAY_CHECK(has_owner)
+      << "Object IDs generated randomly (ObjectID.from_random()) or out-of-band "
+         "(ObjectID.from_binary(...)) cannot be serialized because we do not know their "
```

Comment:
Nice error message but "because we do not know their owner" will likely be very confusing for users. Maybe something like "we aren't able to keep track of their references".

Reply:
Hmm yeah, I went with "we do not know which task will create them". I think this is actually the most accurate.